hoomd/hpmc/update.py

```diff
@@ -631,67 +506,31 @@ def _attach(self):
 
         if not integrator._attached:
             raise RuntimeError("Integrator is not attached yet.")
-        self._cpp_obj = cpp_cls(self._simulation.state._cpp_sys_def,
-                                integrator._cpp_obj,
-                                int(self.seed))
-        super()._attach()
-
-    @property
-    def counter(self):
-        """Get the number of accepted and rejected cluster moves.
 
-        Returns:
-            A counter object with pivot, reflection, and swap properties. Each
-            property is a list of accepted moves and rejected moves since the
-            last run.
-
-        Note:
-            `None` when the simulation run has not started.
-        """
-        if not self._attached:
-            return None
+        if isinstance(self._simulation.device, hoomd.device.GPU):
+            self._cpp_obj = cpp_cls(self._simulation.state._cpp_sys_def,
+                                    integrator._cpp_obj,
+                                    int(self.seed))
         else:
-            return self._cpp_obj.getCounters(1)
-
-    @log(category='sequence')
-    def pivot_moves(self):
-        """tuple[int, int]: Number of accepted and rejected pivot moves.
-
-        Returns:
-            A tuple of (accepted moves, rejected moves) since the last run.
-        """
-        counter = self.counter
-        if counter is None:
-            return (0, 0)
-        else:
-            return counter.pivot
+            self._cpp_obj = cpp_cls(self._simulation.state._cpp_sys_def,
+                                    integrator._cpp_obj,
+                                    int(self.seed))
+        super()._attach()
 
-    @log(category='sequence')
-    def reflection_moves(self):
-        """tuple[int, int]: Number of accepted and rejected reflection moves.
+    @log
+    def avg_cluster_size(self):
+        """float: the typical size of clusters
 
-        Returns:
-            A tuple of (accepted moves, rejected moves) since the last run.
+        0.0 when not attached
```

Comment:
We typically default to `None` when not attached.

Reply:
fixed